app/src/main/java/org/schabi/newpipe/local/bookmark/BookmarkFragment.java

```diff
@@ -26,28 +31,47 @@
 import org.schabi.newpipe.error.ErrorInfo;
 import org.schabi.newpipe.error.UserAction;
 import org.schabi.newpipe.local.BaseLocalListFragment;
+import org.schabi.newpipe.local.holder.LocalBookmarkPlaylistItemHolder;
+import org.schabi.newpipe.local.holder.RemoteBookmarkPlaylistItemHolder;
 import org.schabi.newpipe.local.playlist.LocalPlaylistManager;
 import org.schabi.newpipe.local.playlist.RemotePlaylistManager;
+import org.schabi.newpipe.util.DebounceSavable;
+import org.schabi.newpipe.util.DebounceSaver;
 import org.schabi.newpipe.util.NavigationHelper;
 import org.schabi.newpipe.util.OnClickGesture;
 
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 import icepick.State;
 import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
 import io.reactivex.rxjava3.core.Flowable;
-import io.reactivex.rxjava3.core.Single;
 import io.reactivex.rxjava3.disposables.CompositeDisposable;
 import io.reactivex.rxjava3.disposables.Disposable;
 
-public final class BookmarkFragment extends BaseLocalListFragment<List<PlaylistLocalItem>, Void> {
+public final class BookmarkFragment extends BaseLocalListFragment<List<PlaylistLocalItem>, Void>
+        implements DebounceSavable {
+
+    private static final int MINIMUM_INITIAL_DRAG_VELOCITY = 12;
     @State
     protected Parcelable itemsListState;
 
     private Subscription databaseSubscription;
     private CompositeDisposable disposables = new CompositeDisposable();
     private LocalPlaylistManager localPlaylistManager;
     private RemotePlaylistManager remotePlaylistManager;
+    private ItemTouchHelper itemTouchHelper;
+
+    /* Have the bookmarked playlists been fully loaded from db */
+    private AtomicBoolean isLoadingComplete;
+
+    private DebounceSaver debounceSaver;
```

Comment:
The debounce saver is first introduced in `LocalPlaylistFragment`. Everytime the change occurred, all the videos in the playlist will be deleted and insert again. (This implementation is a bit of silly, I think.) Perhaps deleting 100 records and adding 100 records when the item moves a single row. This will lead to heavy db load.

As for `BookmarkFragment`, the db load is not as heavy as that in `LocalPlaylistFragment`, since it only modifies records it needs to modify. I use the debounce saver in `BookmarkFragment` for consistency. But not using it might be fine.